remotes/docker/resolver.go

```diff
@@ -428,13 +435,15 @@ func (r *dockerBase) setTokenAuth(ctx context.Context, params map[string]string)
 		service: params["service"],
 	}
 
-	scope, ok := params["scope"]
-	if !ok {
-		return errors.Errorf("no scope specified for token auth challenge")
+	if x := ctx.Value(tokenScopesKey{}); x != nil {
+		to.scopes = append(to.scopes, x.([]string)...)
+	}
+	if scope, ok := params["scope"]; ok {
+		to.scopes = append(to.scopes, scope)
+	}
+	if len(to.scopes) == 0 {
+		return errors.Errorf("no token specified for token auth challenge")
```

Comment:
Why the change from scope to token? The term `token` is not correct here in neither case actually since it is a bearer auth challenge.

Reply:
Not sure. Some copy-paste error or typo. :sweat_smile: 
I'll update this PR.